pandas/core/util/hashing.py

```diff
@@ -139,7 +139,10 @@ def hash_pandas_object(
         ser = Series(h, index=obj.index, dtype="uint64", copy=False)
 
     elif isinstance(obj, ABCDataFrame):
-        hashes = (hash_array(series._values) for _, series in obj.items())
+        hashes = (
+            hash_array(series._values, encoding, hash_key, categorize)
```

Comment:
Since `encoding` also wasn't passed before, can you add a test that ensures we respect `encoding` too?

Reply:
I added a `test_df_encoding` to confirm that hash values can change when the different encodings are specified.